lib/chef/resource/rhsm_register.rb

```diff
@@ -117,28 +117,32 @@ class RhsmRegister < Chef::Resource
           end
         end
 
-        execute "Register to RHSM" do
+        register_resource = execute "Register to RHSM" do
```

Comment:
OK so this won't work, because the resource won't have run when you check `updated_by_last_action?` - you're checking at compile time, not run time.

And if you were to call `.run_action(:run)`, you'd change the order of all this which is not what we want.

Instead, I think the right solution is just to make a flush_package_cache resource at the top, with `action :nothing`, and then do immediate notifies to it for these.

Reply:
@jaymzh That used to be true in Chef 16 and Chef 17 (when the resource didn't have unified_mode enabled). That's not true though in Chef 17 when a resource has `unified_mode` enabled or in future Chef 18 where `unified_mode` defaults to true (https://github.com/chef/chef/pull/12692). In other words, there is no longer two phases (compile & converge) in `Chef::Resource` action blocks.

Reply:
Ah I didn't know that made it in.

Nonetheless, you're re-invented `notifies ... :immediately`, so I think we should use that instead. :)